I am not able to build this on ˋx86_64-linuxˋ.
ˋˋˋ
error: builder for '/nix/store/d5qj21wfxvimm05cildaaaylkzl7xri4-zdns-2.0.0.drv' failed with exit code 1;
       last 25 log lines:
       > Running phase: checkPhase
       > --- FAIL: TestConvertNameServerStringToNameServer (0.00s)
       >     --- FAIL: TestConvertNameServerStringToNameServer/Domain_Name_as_Name_Server,_both_IPv4_and_v6 (0.00s)
       >         worker_manager_test.go:56:
       >               Error Trace:    /build/source/src/cli/worker_manager_test.go:56
       >                Error:          Expected nil, but got: &errors.errorString{s:"could not resolve name server: one.one.one.one"}
       >                 Test:           TestConvertNameServerStringToNameServer/Domain_Name_as_Name_Server,_both_IPv4_and_v6
       >     --- FAIL: TestConvertNameServerStringToNameServer/Domain_Name_as_Name_Server,_just_IPv4 (0.00s)
       >         worker_manager_test.go:62:
       >                  Error Trace:    /build/source/src/cli/worker_manager_test.go:62
       >                Error:          Expected nil, but got: &errors.errorString{s:"could not resolve name server: one.one.one.one"}
       >                 Test:           TestConvertNameServerStringToNameServer/Domain_Name_as_Name_Server,_just_IPv4
       >     --- FAIL: TestConvertNameServerStringToNameServer/Domain_Name_as_Name_Server,_just_IPv6 (0.00s)
       >         worker_manager_test.go:68:
       >                 Error Trace:    /build/source/src/cli/worker_manager_test.go:68
       >                Error:          Expected nil, but got: &errors.errorString{s:"could not resolve name server: one.one.one.one"}
       >                 Test:           TestConvertNameServerStringToNameServer/Domain_Name_as_Name_Server,_just_IPv6
       >     --- FAIL: TestConvertNameServerStringToNameServer/Domain_Name_as_Name_Server,_port_provided (0.00s)
       >         worker_manager_test.go:74:
       >                     Error Trace:    /build/source/src/cli/worker_manager_test.go:74
       >                Error:          Expected nil, but got: &errors.errorString{s:"could not resolve name server: one.one.one.one:2345"}
       >                    Test:           TestConvertNameServerStringToNameServer/Domain_Name_as_Name_Server,_port_provided
       > FAIL
       > FAIL   github.com/zmap/zdns/src/cli    0.010s
       > FAIL
ˋˋˋ
I think the tests should be disabled because they need internet access.

--- a/pkgs/by-name/zd/zdns/package.nix
+++ b/pkgs/by-name/zd/zdns/package.nix
@@ -6,22 +6,28 @@
 
 buildGoModule rec {
   pname = "zdns";
-  version = "2023-04-09-unstable";
+  version = "2.0.0";
 
   src = fetchFromGitHub {
     owner = "zmap";
     repo = pname;
-    rev = "ac6c7f30a7f5e11f87779f5275adeed117227cd6";
-    hash = "sha256-que2uzIH8GybU6Ekumg/MjgBHSmFCF+T7PWye+25kaY=";
+    tag = "v${version}";
+    hash = "sha256-ypjUJbTQ0ntLXNRiA96BqDn/SgcbjVe9dd9zmmSqAic=";
   };
 
-  vendorHash = "sha256-daMPk1TKrUXXqCb4WVkrUIJsBL7uzXLJnxWNbHQ/Im4=";
+  vendorHash = "sha256-Q7W+G/yA/EvHqvDUl3T5BoP+K5ZTcLFLSaWw4TPMH2U=";
 
-  meta = with lib; {
+  preCheck = ''
+    # Tests require network access
+    substituteInPlace src/cli/worker_manager_test.go \
+      --replace-fail "TestConvertNameServerStringToNameServer" "SkipTestConvertNameServerStringToNameServer"
+  '';
+
+  meta = {
     description = "CLI DNS lookup tool";
     mainProgram = "zdns";
     homepage = "https://github.com/zmap/zdns";
-    license = with licenses; [ asl20 ];
-    maintainers = with maintainers; [ fab ];
+    license = with lib.licenses; [ asl20 ];
+    maintainers = with lib.maintainers; [ fab ];
   };
 }

